Feature/vcpkg linux

sylvlecl opened the pull request:
Continuing work on VCPKG for Linux, basically trying to simplify workflows from https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2049 and using manifest mode.

**Status**
- Build OK for all workflows
- Caching OK for all workflows

**Overall strategy**

Merge this in feature branch `feature/vcpkg-dependencies` first.
The whole feature branch will be merged only once it brings sufficient added value to existing mechanisms.

In particular, when we are able to remove the use of `antares-deps` in favor of using vcpkg would be a good time to do so.

Documentation should be updated in `feature/vcpkg-dependencies` when the developer workflow is clear.

sylvlecl committed:
VCPKG manifest mode with minizip and boost only, ubuntu

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Fix VCPKG install

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
To be reverted: disable temporarily non-ununtu CIs

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Fix toolchain file path

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Fix wxWidgets search

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Revert boost from vcpkg for now

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Save cache even when build fails

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Fix cache key

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Restore windows build

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Fix vcpkg install

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try fixing toolchain path

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try fiwing workspace path

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Another try

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try improving cache logic

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Add boost to windows build

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try with centos

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
No VCPKG cache for centos, for now

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try to fix path to vcpkg

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Log for debugging

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try install cmake with pip instead

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
More logs

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Temporarily disable ubuntu & windows

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try fix path

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Restore gcc install

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Again another try

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Fix git version

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Fix packaging too

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try oracle 8

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Fix security error

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Revert "Temporarily disable ubuntu & windows"

This reverts commit 5e26bbda6b09772fd5587f78b46a7f7c6d00cd1e.

sylvlecl committed:
Hashing after submodule update

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Fix restore keys

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Disable ubuntu & windows

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try cache action downgraded

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try cache action v3 on centos7

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Add actual caching

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Try env vars

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Skip some tests for testing

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
fix

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
fix

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Revert "Skip some tests for testing"

This reverts commit ff0d26182832f60f8baceaaefbb73f2ea33df40b.

sylvlecl committed:
Fix vcpkg cache paths on centos

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Revert "Disable ubuntu & windows"

This reverts commit 0a52474088c0ecef572f5a119d6c2fc1c69a6c56.

sylvlecl committed:
Disable centos and oracle on PRs

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Merge remote-tracking branch 'origin/develop' into feature/vcpkg-linux

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Restore sonarcloud

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Fix windows workflow file

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
VCPKG for sonar job

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

a-zakir commented:
I could not agree more

sylvlecl committed:
Support minizip dependency as minizip or minizip-ng

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Merge branch 'feature/vcpkg-dependencies' into feature/vcpkg-linux

flomnes commented on .github/workflows/centos7.yml:
```diff
@@ -139,4 +161,10 @@ jobs:
-  
\ No newline at end of file
+    - name: Cache vcpkg binary dir
+      if: always()
```
Even in the case of build errors ?

sylvlecl replied:
Good question :
I think it makes sense:
- if the dependencies build works and the project build fails, we need to keep the dependencies to not rebuild them in a later workflow
- if it's the build of a dependency which fails, I think it is still better to save the cache because other dependencies may have succeeded. Also, because we restore the previous cache, this should keep previously cached items which have not been rebuilt (like the ones that would have been rebuilt after the dependency that failed)

flomnes committed:
Empty commit to test GH caching

sylvlecl committed:
Add custom triplets to:
 - link statically with chosen libraries
 - build only release artifacts on CI

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl commented:
**To be checked before merging:**

I am not sure the windows artifact will be complete now, because with the default triplet "x64-windows", dependencies are built as DLLs, which will include zlib and minizip. Currently, they are linked statically.

I would propose to introduce custom triplets for this, and also to avoid building debug artifacts of dependencies.

sylvlecl committed:
Fix build type in triplets

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

flomnes commented on src/triplets/x64-linux-antares.cmake:
remove ?

sylvlecl committed:
Fix windows triplets

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

a-zakir commented on .github/workflows/centos7.yml:
```diff
@@ -43,7 +43,28 @@ jobs:
+        path: ${{ env.VCPKG_CACHE_DIR }}
+        key: vcpkg-cache-centos7-${{ hashFiles('src/vcpkg.json', '.git/modules/vcpkg/HEAD') }}
+        # Allows to restore a cache when deps have only partially changed (like adding a dependency)
+        restore-keys: vcpkg-cache-centos7-
```
wow! i would recommend to place the "install vcpkg and the restore cache" jobs in dedicated file

sylvlecl replied:
Unfortunately it does not seem so easy, because we cannot have "post actions" in composite github actions :( 
See https://github.com/actions/runner/issues/1478

I wanted to have the 3 steps "install VCPKG", "restore cache" and "save cache" in the same action, but it's not possible.
If we keep the save step separate, it will be messy because you need to know "internal details" of the other steps (the cahce key basically).

So I'm afraid I will leave it this way for now ...

a-zakir commented on src/vcpkg.json:
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "antares-simulator",
+  "version-string": "9.1.0",
```
i don't know if it's possible, but i will be nice to update the version automatically

a-zakir commented on src/packaging/Config.cmake.in:
```diff
@@ -1,8 +1,16 @@
+else ()
+	find_dependency(minizip)
+	if (NOT minizip_FOUND)
+		message (FATAL_ERROR "Minizip dependency (minizip or minizip-ng) not found.")
```
specify minimal version? according to this [commit](https://github.com/AntaresSimulatorTeam/Antares_Simulator/commit/27de5803089987b907cd136d4277a9d53bd6040c#diff-d0264d71f2ec4efa0918b170f9e350b879ab24b695165a9f41358e74bf2b8fe8), at least v 3.0.7 and back are incompatible with Simulator.
minizip target is confusing it actually refers to [minizip-ng](https://github.com/zlib-ng/minizip-ng?tab=readme-ov-file) and not the ["original"](https://github.com/madler/zlib/tree/master/contrib/minizip) project

a-zakir commented on src/tests/src/libs/antares/writer/CMakeLists.txt:
```diff
@@ -8,7 +8,7 @@ target_link_libraries(test-writer
 		Antares::result_writer
 		test_utils_unit
-		MINIZIP::minizip
+		MINIZIP::minizip-ng
```
is this  works with the "minizip" target ?

JasonMarechal25 commented on .github/workflows/centos7.yml:
```diff
@@ -43,7 +43,28 @@ jobs:
+    - name: Install cmake 3.28
+      run: pip3 install cmake==3.28.4
+
+    - name: Install VCPKG
```
For a v2 maybe use https://github.com/lukka/run-vcpkg

JasonMarechal25 committed:
Boost available on linux too

JasonMarechal25 committed:
Merge branch 'develop' into feature/vcpkg-linux

JasonMarechal25 committed:
Feature/vcpkg linux fixes (#2127)

- Define  minimal version for boost and minizip
- remove BOOST_TEST_DYN_LINK

---------

Co-authored-by: Florian OMNES <florian.omnes@rte-france.com>

JasonMarechal25 committed:
Merge branch 'develop' into feature/vcpkg-linux
